### ISSUE TITLE
Fatal Error: Failed to execute script main

### ISSUE THREAD

**Issue by mattling9** [pyinstaller/pyinstaller#3446]
I'm running PyInstaller 3.4.dev0+956066103 on Python 3.4.4 with PyQt4. I'm running Windows 7.6.17601-SP1.

The .exe is compiled without any issues, however when i try and run the .exe file i get the following error: Fatal Error: Failed to execute script main

I tried the following with no success:
- Copying Crypto folder in Python34/Lib/site-packages and renaming it to Cryptodome.
- changing all imports in site-packages/Crypto/Util/_raw_api.py from Crypto to Cryptodome.

I'v tried removing --onefile and adding --debug but this does not help the issue.

Any suggestions?

**Comment by htgoebel** [comment]
Please provide more details, see [How to report Bugs](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs). Esp. please provide a *minimal* example to reproduce the error.

**Comment by AmjadHD** [comment]
I have a similar issue:

```shell
C:\Users\User\test\Shmup Folder 2\dist>Shmup
Traceback (most recent call last):
  File "Shmup.py", line 4, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "c:\users\user\test\lib\site-packages\PyInstaller\loader\pyimod03_importe
rs.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "initialisation.py", line 28, in <module>
  File "spritesheet.py", line 6, in __init__
pygame.error: Couldn't open C:\Users\User\AppData\Local\Temp\_MEI50642\images\sh
ipsheet.png
[2948] Failed to execute script Shmup

C:\Users\User\test\Shmup Folder 2\dist>
```

i don't know why pygame searches Temp dir although it was working fine before i turn the script into executable ?

**Comment by tallforasmurf** [comment]
The error is that pygame wants something called `shipsheet.png` and that is not in the bundle. Probably a data file, see [adding data files](https://pyinstaller.readthedocs.io/en/stable/spec-files.html#adding-data-files).
To answer your other question, see [how it works](https://pyinstaller.readthedocs.io/en/stable/operating-mode.html#how-the-one-file-program-works).

**Comment by htgoebel** [comment]
@mattling9 I've just seen, this is about Cryptodome. Please try #3424 

https://github.com/pyinstaller/pyinstaller/wiki/How-to-apply-a-Patch-or-Pull-Request-to-PyInstaller

**Comment by htgoebel** [comment]
@mattling9 Update: I just merged #3424, sou you can simply use the current development head.

**Comment by mattling9** [comment]
@htgoebel Went to my PyInstaller folder and had no file named hook-Crypto, only hook-Cryptodome. I created a .py file under the name hook-Crypto and copied and pasted the hook suggested. This solved my problem!

Thank you very much for your help, i really appreciate it!
